Ian/nlb

Bellardia opened the pull request:
Add an Nginx mod for rule:
```
real_ip_header proxy_protocol_xfwd;
```
which is a cascading rule of `proxy_protocol` followed by `X-Forwarded-For`.

Also addresses some bad config for openssl-source that makes it impossible to build nginx after /tmp has been cleared.

Bellardia committed:
Support for proxy_protocol

Bellardia committed:
Fix openssl build to have proper dest dir

Bellardia committed:
Don't specify openssl path

Bellardia committed:
Ensure domain is not empty

ahosking reviewed (approved):
Well done. It's relatively clean for the headache!

Bellardia committed:
Link against openssl libs

Bellardia committed:
8443 listeners to have ssl

Bellardia committed:
Enable port reuse on ssl

Bellardia committed:
Manage process_control_timeout

Bellardia committed:
Fix regex bug on replacement pattern

Bellardia committed:
Implement more predictable credential sync patterns for JBX

Bellardia committed:
Implement more predictable credential sync patterns for Admin

Bellardia committed:
Remove unnecessary php reloads

Bellardia committed:
Further increase opcache max_file_size

Bellardia committed:
Add commands for elb registration

Bellardia committed:
Disable fast shutdown

From https://www.php.net/manual/en/opcache.configuration.php#ini.opcache.fast-shutdown

This directive has been removed in PHP 7.2.0. A variant of the fast shutdown sequence has been integrated into PHP and will be automatically used if possible.

Bellardia committed:
Fix typo in attribute check

Bellardia committed:
Add ohai plugin for opsworks

Bellardia committed:
Increased allowed # of cached files

Bellardia committed:
Ensure env is set in deploy script

Bellardia committed:
Fix hosts recipe to work with chef stacks 12

Bellardia committed:
Disable opsworks_stack_state_sync

Bellardia committed:
Disable xdebug extension in non dev envs

Bellardia committed:
Add max_requests and dd trace options

Bellardia committed:
true/false needs to be 0/1

Bellardia committed:
Correct action from disabled => disable

Bellardia committed:
Add node address to definition

Bellardia committed:
Leave consul cluster when instance shuts down

Bellardia committed:
Auto register elb in configure

Bellardia committed:
Set datadog hostname more reliably

Bellardia committed:
Sync admin nginx conf with old prod

Bellardia committed:
Return 404 for robots/favicon

Bellardia committed:
Fix bug where class is counted instead of workers

Bellardia committed:
Fix gearman service to correctly clean up on stop

Bellardia committed:
Remove https redirect

Bellardia committed:
Improve gearman recipes

Bellardia committed:
Ensure service file installs correctly

Bellardia committed:
Set specific commit hash

Bellardia committed:
Optimize gearman, install via chef

Bellardia committed:
Allow planting user-provided credential overrides

Bellardia committed:
Use hostname as table name if none is specified

Bellardia committed:
Use Timescale 2 / Postgres 13

Bellardia committed:
Drop db after CI

Bellardia committed:
Purge postgres12 loader

Bellardia committed:
Backup nvme mysql data periodically

Bellardia committed:
Stop postgres when removing old version

Bellardia committed:
Disable mysql logs

Bellardia committed:
Ensure bak is defined for all envs

Bellardia committed:
Datadog CI integration

Bellardia committed:
Don't bind ssl if cert is not available

Bellardia committed:
Add default route for thisthingisnotathing

Bellardia committed:
Update to node 14

Bellardia committed:
Add cookbook for installing our blog

Bellardia committed:
Use variable for upstream name

Bellardia committed:
Move to package cloud

Bellardia committed:
Don't attempt to check for ip based domains

Bellardia committed:
Use loggable flag on access_logs

Bellardia committed:
Add logs integration

Bellardia committed:
Use timescale ppa2 for timescale 2

Bellardia committed:
Reduce DB memory usage

Bellardia committed:
Build pgloader from source for postgres13 compat

Bellardia committed:
Add TLS 1.3, optimize nginx ssl conf

Bellardia committed:
Optimize sysctl params

Bellardia committed:
Remove duplicate dhparam

Bellardia committed:
Support setting initcwnd and initrwnd

Bellardia committed:
Run seeds as www-data

Bellardia committed:
Further reduce per-conn limits

Bellardia committed:
Disable performance schema

Bellardia committed:
Move seed status to root execution
